pkgs/applications/virtualization/xen/generic.nix

```
@@ -1,265 +1,376 @@
+        #!${stdenv.shell} -e
+        echo ===== FAKE WGET: Not fetching \$*
+        [ -e \$3 ]
+        EOF
+
+        cat > fake-bin/git << EOF
+        #!${stdenv.shell}
+        echo ===== FAKE GIT: Not cloning \$*
+        [ -e \$3 ]
+        EOF
+
+        chmod +x fake-bin/*
+        export PATH=$PATH:$PWD/fake-bin
+      ''
+
+      # Remove in-tree QEMU sources, as we either pre-fetch them through
+      # the versioned Nix expressions if withInternalQEMU is true, or we
+      # don't build QEMU at all if withInternalQEMU is false.
+      + ''
+        rm -rf tools/qemu-xen tools/qemu-xen-traditional
+      ''
+
+      # The following expression moves the sources we fetched in the
+      # versioned Nix expressions to their correct locations inside
+      # the Xen source tree.
+      + ''
+        ${withPrefetchedSources (
+          name: pkg: ''
+            echo "Copying pre-fetched source: ${pkg.src} -> tools/${name}"
+            cp -r ${pkg.src} tools/${name}
+            chmod -R +w tools/${name}
+          ''
+        )}
+      '';
+
+    patches = config.patches or [ ];
+
+    postPatch =
+      # The following patch forces Xen to install xen.efi on $out/boot
+      # instead of $out/boot/efi/efi/nixos, as the latter directory
+      # would otherwise need to be created manually.
+      ''
+        substituteInPlace xen/Makefile \
+          --replace-fail "\$(D)\$(EFI_MOUNTPOINT)/efi/\$(EFI_VENDOR)/\$(T)-\$(XEN_FULLVERSION).efi" \
+                    "\$(D)\$(BOOT_DIR)/\$(T)-\$(XEN_FULLVERSION).efi"
+      ''
+
+      # The following patch fixes the call to /bin/mkdir on the
+      # launch_xenstore.sh helper script.
+      + ''
+        substituteInPlace tools/hotplug/Linux/launch-xenstore.in \
+          --replace-fail "/bin/mkdir" "${coreutils}/bin/mkdir"
+      ''
+
+      # The following expression fixes the paths called by Xen's systemd
+      # units, so we can use them in the NixOS module.
+      + ''
+        substituteInPlace \
+          tools/hotplug/Linux/systemd/{xen-init-dom0,xen-qemu-dom0-disk-backend,xenconsoled,xendomains,xenstored}.service.in \
+          --replace-fail /bin/grep ${gnugrep}/bin/grep
+        substituteInPlace \
+         tools/hotplug/Linux/systemd/{xen-qemu-dom0-disk-backend,xenconsoled}.service.in \
+          --replace-fail "/bin/mkdir" "${coreutils}/bin/mkdir"
+      ''
+
+      # The following expression applies the patches defined on each
+      # prefetchedSources attribute.
+      + ''
+        ${withTools "patches" (
+          name: pkg: ''
+            ${lib.strings.concatMapStringsSep "\n" (patch: ''
+              echo "Patching with ${patch}"
+              patch -p1 < ${patch}
+            '') pkg.patches}
+          ''
+        )}
+
+             ${withTools "postPatch" (name: pkg: pkg.postPatch)}
+
+             ${config.postPatch or ""}
+      '';
+
+    makeFlags = [
+      "XSM_ENABLE=y"
+      "FLASK_ENABLE=y"
+
+      "EFI_VENDOR=nixos"
```

```suggestion
      "EFI_VENDOR=nixpkgs"
```

`nixos` seems inaccurate as this might get called on a standalone nix

---

No, this should be kept at whatever the value of `nixosDir` in [systemd-boot.nix](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix#L51) is. (removing the `/EFI/` prefix)

---

Make this use the `nixosDir` variable instead then?

---

That's impossible due to sandboxing. You can't access NixOS modules in nixpkgs derivations.

---

> it's not related to sandboxing but there would be no convenient or well-layered way to access that specific definition and the code is fine as-is -- Nixpkgs developer

I guess it's okay then